### ISSUE TITLE
Set direction of the text automatically (For RTL text)

### ISSUE THREAD

**Issue by mokazemi** [ueberdosis/tiptap#3085]
### What problem are you facing?

I can't use the editor for Right-to-Left languages, because there's no option for it. 
So the editor is only limited to English and other Left-to-Right languages.

### What’s the solution you would like to see?

There are some ways to set text direction automatically. For example this area that I'm writing in (Github text area!) has this feature.
So when I write something in a RTL languages, it's displayed correctly.

![image](https://user-images.githubusercontent.com/18334056/184470465-7449a140-bb4c-4ce9-b96d-0743e5999813.png)

![image](https://user-images.githubusercontent.com/18334056/184470527-090be258-5c58-4b21-9252-78ff9d225c2b.png)

### What alternatives did you consider?

There could at least be a button for it like some Word Processors such as Google Docs, like this: 
![image](https://user-images.githubusercontent.com/18334056/184470599-c48f70e0-44da-485b-b35c-c33f621984ef.png)

But some text editors such as Github text area doesn't have it, but set the direction automatically. 

### Anything to add? (optional)

I'm facing this problem in the Nextcloud Editor

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by mustafa-online** [comment]
This is the only missing part that keeping me from using Tiptap.

**Comment by mustafa-online** [comment]
+1

**Comment by hadifarnoud** [comment]
+1

**Comment by avivdaniel** [comment]
+1

**Comment by mustafa-online** [comment]
Any updates?

**Comment by amirhhashemi** [comment]
This is what you are looking for: https://www.npmjs.com/package/tiptap-text-direction

**Comment by bdbch** [comment]
@mokazemi @amirhhashemi yes we (or at least I am) inexperienced with RTL languages and I think I personally could improve here.

Could you point me to the PR regarding this issue? I'd be interested if we can implement support without breaking existing Extensions.

**Comment by amirhhashemi** [comment]
> Could you point me to the PR regarding this issue? I'd be interested if we can implement support without breaking existing Extensions.

This is the original PR: https://github.com/ueberdosis/tiptap/pull/2963
It's a little out-dated but I can update if you are interested. I haven't tested it in a real project yet but theoretically it shouldn't break any existing extension.

If you want to test the latest version checkout this repo:  https://github.com/amirhhashemi/tiptap-text-direction. It's pretty much the same thing but I fixed some bugs and added more stuff.

**Comment by yousefhalabi** [comment]
@amirhhashemi thanks for the extension, i can confirm it works on vue3 and saved me a lot of the trouble cheers!

**Comment by coderyeasin** [comment]
@amirhhashemi Thank you! for developing this man, I'm facing some problems with this plugin and I'm using nextjs. 
I can't write Arabic (so I don't know if when writing Arabic, it changes direction or not) , but when I paste something inline it will take RTL and click enter if I paste again Arabic text it will also take RTL, so kind of good for that but the real problem is when i paste multiple Arabic lines its make only first-line direction(RTL), others line are taking LTR. so how can I resolve it with styles that can you explain please see this pic 
![Screenshot from 2023-04-13 01-46-12](https://user-images.githubusercontent.com/33999408/231571429-246af386-2f01-427e-b5af-118219ce2ccb.png)

**Comment by amirhhashemi** [comment]
@coderyeasin Thank you for reporting this. I opened an issue in the main repo: https://github.com/amirhhashemi/tiptap-text-direction/issues/8

Please refer to that for further follow-up
